File: src/main/java/destiny/core/calendar/decorators/DayDecorator.java
/**
 * @author smallufo
 * Created on 2008/4/9 at 上午 11:28:28
 */
package destiny.core.calendar.decorators;

import com.google.common.collect.ImmutableMap;
import destiny.tools.Decorator;
import destiny.tools.LocaleUtils;
import org.jetbrains.annotations.NotNull;

import java.util.Locale;

public class DayDecorator {

  private final static ImmutableMap<Locale, Decorator<Integer>> implMap = new ImmutableMap.Builder<Locale, Decorator<Integer>>()
    .put(Locale.CHINESE, new DayDecoratorChinese())
    .put(Locale.ENGLISH, new DayDecoratorEnglish())
    .build();

  @NotNull
  public static String getOutputString(int value, Locale locale) {
    return implMap.get(
      LocaleUtils.getBestMatchingLocale(locale, implMap.keySet()).orElse((Locale) implMap.keySet().toArray()[0])
    ).getOutputString(value);
  }
}

File: src/main/kotlin/destiny/core/calendar/decorators/DayDecorator.kt
/**
 * @author smallufo
 * Created on 2008/4/9 at 上午 11:28:28
 */
package destiny.core.calendar.decorators

import destiny.tools.Decorator
import destiny.tools.getOutputString
import java.io.Serializable
import java.util.*


object DayDecorator {

  private val implMap = mapOf<Locale , Decorator<Int>>(
    Locale.CHINESE to DayDecoratorChinese(),
    Locale.ENGLISH to DayDecoratorEnglish()
  )

  fun getOutputString(value: Int, locale: Locale): String {
    return implMap.getOutputString(value , locale)
  }
}


class DayDecoratorChinese : Decorator<Int>, Serializable {
  override fun getOutputString(value: Int): String {
    return value.toString() + "日"
  }
}

class DayDecoratorEnglish : Decorator<Int>, Serializable {

  override fun getOutputString(value: Int): String {
    return value.toString()
  }

}
